openml/datasets/functions.py

```diff
@@ -609,40 +667,86 @@ def _create_dataset_from_description(description, features, qualities, arff_file
-        description.get("oml:row_id_attribute"),
-        description.get("oml:ignore_attribute"),
-        description.get("oml:version_label"),
-        description.get("oml:citation"),
-        description.get("oml:tag"),
-        description.get("oml:visibility"),
-        description.get("oml:original_data_url"),
-        description.get("oml:paper_url"),
-        description.get("oml:update_comment"),
-        description.get("oml:md5_checksum"),
+        data_format=description["oml:format"],
+        dataset_id=description["oml:id"],
+        version=description["oml:version"],
+        creator=description.get("oml:creator"),
+        contributor=description.get("oml:contributor"),
+        collection_date=description.get("oml:collection_date"),
+        upload_date=description.get("oml:upload_date"),
+        language=description.get("oml:language"),
+        licence=description.get("oml:licence"),
+        url=description["oml:url"],
+        default_target_attribute=description.get(
+            "oml:default_target_attribute"
+        ),
+        row_id_attribute=description.get("oml:row_id_attribute"),
+        ignore_attribute=description.get("oml:ignore_attribute"),
+        version_label=description.get("oml:version_label"),
+        citation=description.get("oml:citation"),
+        tag=description.get("oml:tag"),
+        visibility=description.get("oml:visibility"),
+        original_data_url=description.get("oml:original_data_url"),
+        paper_url=description.get("oml:paper_url"),
+        update_comment=description.get("oml:update_comment"),
+        md5_checksum=description.get("oml:md5_checksum"),
         data_file=arff_file,
         features=features,
-        qualities=qualities)
+        qualities=qualities,
+    )
     return dataset
+
+
+def _get_online_dataset_arff(dataset_id):
+    """Download the ARFF file for a given dataset id
+    from the OpenML website.
+
+    Parameters
+    ----------
+    dataset_id : int
+        A dataset id.
+
+    Returns
+    -------
+    str
+        A string representation of an ARFF file.
+    """
+    dataset_xml = openml._api_calls._perform_api_call("data/%d" % dataset_id)
+    # build a dict from the xml.
+    # use the url from the dataset description and return the ARFF string
+    return openml._api_calls._read_url(
+        xmltodict.parse(dataset_xml)['oml:data_set_description']['oml:url']
+    )
+
+
+def _get_online_dataset_format(dataset_id):
+    """Get the dataset format for a given dataset id
+    from the OpenML website.
+
+    Parameters
+    ----------
+    dataset_id : int
+        A dataset id.
+
+    Returns
+    -------
+    str
+        Dataset format.
+    """
+    dataset_xml = openml._api_calls._perform_api_call("data/%d" % dataset_id)
+    # build a dict from the xml and get the format from the dataset description
+    return xmltodict\
+        .parse(dataset_xml)['oml:data_set_description']['oml:format']\
+        .lower()
```

Comment:
https://github.com/openml/openml-python/blob/e711267cc4acdfa82610e880fb7307ec9959a2bd/tests/test_datasets/test_dataset_functions.py#L468-L477

https://github.com/openml/openml-python/blob/e711267cc4acdfa82610e880fb7307ec9959a2bd/tests/test_datasets/test_dataset_functions.py#L511-L520

@mfeurer Now that I am thinking about it, the above checks are happening in test_create_dataset_sparse and test_create_dataset_list. Although they check that the datasets are uploaded correctly, they also check that the functions work correctly. In my opinion this already covers the functions, any other test where we do not upload, in the future might fail if information is removed from the test server.

Reply:
True, but this also suboptimal because it is not immediately clear where this is tested. How about a simple test like for all the other functions which retrieve dataset related stuff?

Reply:
I agree that it is not clear where it is tested.. Are you thinking of checking an existing dataset on the server ?